src/components/Navbar.js

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+const handleTitleBlur = async () => {
+  setIsLoading(true);
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
setIsLoading(true);
try {
  // existing logic
} catch (error) {
  setIsLoading(false);
  // error handling
}
```
Consider adding error handling to reset the loading state when an error occurs. Currently, setIsLoading(true) is set but there's no visible error state management in the provided code block.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;